File: NRZ.Web/client/src/models/gridSearchModel.ts
import { IGridSearchModel } from '@/interfaces/IGridSearchModel';

export default class GridSearchModel implements IGridSearchModel {
    constructor(search: string, sortBy: string, sortDesc: boolean, page: number, itemsPerPage: number, startDate: Date, endDate: Date) {
        this.search = search,
        this.sortBy = sortBy,
        this.sortDesc = sortDesc,
        this.page = page,
        this.itemsPerPage = itemsPerPage
        this.startDate = startDate,
        this.endDate = endDate;
    }
    search: string;
    sortBy: string;
    sortDesc: boolean;
    page: number;
    itemsPerPage: number;
    startDate: Date;
    endDate: Date;
}

File: NRZ.Models/GridModels/GridSearchModel.cs
using System;

namespace NRZ.Models.GridModels
{
    public class GridSearchModel
    {
        public string Search { get; set; }

        public string SortBy { get; set; }

        public bool SortDesc { get; set; }

        public int Page { get; set; }

        public int ItemsPerPage { get; set; }   

        public DateTime? StartDate { get; set; }

        public DateTime? EndDate { get; set; }
    }
}
